enhancements/ingress/lb-allowed-source-ranges.md

```diff
@@ -0,0 +1,292 @@
+2. Adding `service.beta.kubernetes.io/load-balancer-source-ranges` annotation to 
+a LoadBalancer Service and observing cluster's upgradable status.
+3. Setting `.Spec.LoadBalancerSourceRanges` on the LoadBalancer Service 
+when `AllowedSourceRanges` is empty and observing cluster's upgradable status.
+
+This enhancement adds an upgrade test in openshift/origin to make sure that 
+the migration workflow explained under **Variation** above works seamlessly.
+In detail, it implements 
+[Test](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/upgrades/upgrade.go#L46) 
+interface that includes the following steps:
+
+1. In `Setup`, create an IngressController that requests an LB, 
+and set the `service.beta.kubernetes.io/load-balancer-source-ranges` annotation
+on the resulting LoadBalancer-type service.
+2. In `Test`, periodically check the service to verify that nothing updates the annotation
+or the `Spec.LoadBalancerSourceRanges` field on the service during the upgrade.
+3. In `Test`, verify again when the upgrade is done that nothing has updated the annotation
+or `Spec.LoadBalancerSourceRanges`, and verify that the operator has set `Upgradeable=False`.
+4. Then set `AllowedSourceRanges`, remove the annotation, and finally verify that the operator 
+has set `Upgradeable=True` and `Spec.LoadBalancerSourceRanges` accordingly.
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A. This feature will go directly to GA.
+
+#### Tech Preview -> GA
+
+N/A. This feature will go directly to GA.
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+The `AllowedSourceRanges` field is empty by default, so it will be empty on an upgraded cluster, until the cluster admin 
+sets it after the upgrade. If the cluster admin sets `AllowedSourceRanges` and then downgrades the cluster, 
+the `AllowedSourceRanges` field will vanish from the CRD; however, the downgraded operator would not update or 
+remove the `Spec.LoadBalancerSourceRanges` field value and would not update or remove the `service.beta.kubernetes.io/load-balancer-source-ranges`
+annotation. In any case, none of this (not `AllowedSourceRanges` on the IngressController, nor `LoadBalancerSourceRanges` or 
+`service.beta.kubernetes.io/load-balancer-source-ranges` on the service) should have any effect on any deployment.
+
+Additionally, the cluster will be marked as non-upgradable:
+- If `service.beta.kubernetes.io/load-balancer-source-ranges` annotation is set 
+to guide the cluster admin towards using the IngressController API and deprecate
+use of the service annotation for ingress.
+- If `Spec.LoadBalancerSourceRanges` does not match `AllowedSourceRanges` on the 
+IngressController. This can be the case when the cluster admin has configuration in 
+`Spec.LoadBalancerSourceRanges` prior to upgrading to OpenShift 4.12 that is not overwritten 
+by `AllowedSourceRanges` as it is empty in the beginning.
```

Comment:
```suggestion
- If `Spec.LoadBalancerSourceRanges` on the service is nonempty and `AllowedSourceRanges` on the 
IngressController is empty. This can be the case when the cluster admin has configuration in 
`Spec.LoadBalancerSourceRanges` prior to upgrading to OpenShift 4.12 that is not overwritten 
by `AllowedSourceRanges` as it is empty in the beginning.
```
Isn't this what we agreed on in discussions with @deads2k and what the table in "Implementation Details/Notes/Constraints" shows?

Reply:
It is. Thanks for making it more clear.